embeddings/data/data_loader.py

```diff
@@ -74,9 +74,13 @@ def load(self, dataset: Dataset[T_path]) -> Corpus:
         )
 
 
-def get_hf_dataloader(
-    dataset: HuggingFaceDataset,
-) -> Union[HuggingFaceDataLoader, HuggingFaceLocalDataLoader]:
+HF_DATALOADERS = Union[HuggingFaceDataLoader, HuggingFaceLocalDataLoader]
+FLAIR_DATALOADERS = Union[
+    HuggingFaceDataLoader, PickleFlairCorpusDataLoader, ConllFlairCorpusDataLoader
+]
+
+
+def get_hf_dataloader(dataset: LoadableDataset) -> HF_DATALOADERS:
```

Comment:
how much work would it take to create a `AutoDataLoader` class (such as the ones for embedding)? do you think it would be a good idea?

Reply:
I think that's not so easy, because we would need to add functionality that detects whether given dataset is hf datasets or flair datasets. Without that it would be not so `auto`